plugins/modules/infra_envs.py

```diff
@@ -23,48 +25,126 @@
+        description: SSH public key for debugging the installation
+        required: false
+        type: str
+    cluster_id:
+        description: The cluster ID to associate with the infra-env
         required: false
         type: str
+    image_type:
+        description: The type of image to generate
+        required: false
+        type: str
+        choices: ["full-iso", "minimal-iso"]
+        default: "minimal-iso"
+    proxy:
+        description: Proxy settings for the infra-env
+        required: false
+        type: dict
+        suboptions:
+            http_proxy:
+                description: HTTP proxy URL
+                type: str
+            https_proxy:
+                description: HTTPS proxy URL
+                type: str
+            no_proxy:
+                description: Comma-separated list of hosts that should not use proxy
+                type: str
+    static_network_config:
+        description: Static network configuration for the infra-env
+        required: false
+        type: list
+        elements: dict
+    ignition_config_override:
+        description: Ignition config override for the infra-env
+        required: false
+        type: str
+    kernel_arguments:
+        description: Additional kernel arguments for the infra-env
+        required: false
+        type: list
+        elements: dict
 
 author:
     - Vishwanath Jayaraman (@vjayaramrh)
-
 """
 
 EXAMPLES = r"""
-# Use argument
-- name: create infra-envs
+# Create infra-env
+- name: Create infra-env
   infra_envs:
     state: present
-    name: infra1
-    pull_secret: blah
-
+    name: my-infra-env
+    pull_secret: "{{ pull_secret }}"
+    openshift_version: "4.19"
+    cpu_architecture: "x86_64"
+    ssh_authorized_key: "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQ..."
+    image_type: "minimal-iso"
+    proxy:
+      http_proxy: "http://proxy.example.com:8080"
+      https_proxy: "https://proxy.example.com:8080"
+      no_proxy: "localhost,127.0.0.1"
+
+# List all infra-envs
 - name: List infra-envs
-  infra_envs: {}
+  infra_envs:
+  register: infra_envs_list
 
-- name: Retrieve details of a specific infra-envs resource
+# Get details of a specific infra-env
+- name: Get infra-env details
   infra_envs:
     infra_env_id: "deadbeef-dead-beef-dead-beefdeadbeef"
+  register: infra_env_details
 
-- name: Delete infra-envs
+# Update infra-env
+- name: Update infra-env
+  infra_envs:
+    state: present
+    infra_env_id: "deadbeef-dead-beef-dead-beefdeadbeef"
+    ssh_authorized_key: "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQ..."
+    proxy:
+      http_proxy: "http://newproxy.example.com:8080"
+
+# Delete infra-env
+- name: Delete infra-env
   infra_envs:
     state: absent
     infra_env_id: "deadbeef-dead-beef-dead-beefdeadbeef"
 """
 
 RETURN = r"""
-result:
-    description: Result from the API POST call
-    type: dict
+infra_envs:
+    description: List of infra-envs or single infra-env details
+    type: list or dict
```

Comment:
`list_infra_envs` returns a list but other methods return a dict. Not sure what to do about this error...
```
  ERROR: plugins/modules/infra_envs.py:0:0: return-syntax-error: RETURN.infra_envs.type: not a valid value for dictionary value @ data['infra_envs']['type']. Got 'list or dict'
```
Maybe refactor to always return a list to keep it consistent?